src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -113,12 +111,14 @@ internal static bool DoSslHandshake(IntPtr sslContextPtr, IntPtr recvPtr, int re
             }
 
             int error;
+
             if (retVal != 1)
             {
-                error = GetSslError(context.sslPtr, retVal, "SSL Handshake failed: ");
+                error = GetSslError(context.sslPtr, retVal);
+
                 if ((retVal != -1) || (error != libssl.SslErrorCode.SSL_ERROR_WANT_READ))
                 {
-                    throw CreateSslException(context.sslPtr, "SSL Handshake failed: ", error);
+                    throw CreateSslException(context.sslPtr, "SSL Handshake failed: ", retVal);
```

Comment:
This is another example where this will need to be in the .resx file.  Most likely the string would end up being something like "SSL handshake failed with error {0}", and then the call site would use SR.Format to generate the string to pass to the exception.